### DISCUSSION TITLE
What impact does having {% load static %} in a component have on rendering performance?

### DISCUSSION THREAD

**Discussion by ra-dave** [wrabit/django-cotton#172] [General]
I have a tooltip component that uses {% load static %}.  I have dozens of them on a single web page.  Is that going to impact rendering time or does caching make it a non-factor other than maybe the first time?

**Comment by wrabit** [comment]
If you're literally referring to using the `{% load static %}`, I don't think it's anything to worry about. It just something similar to importing a module. But if you mean using `{% static 'some.js' %} pulling in the same asset each time, say a js file then that may have some consequences. i.e. if it's a library to implement the tooltips functionality, that may be better placed outside of the component.

**Comment by ra-dave** [comment]
Yes, it would be used to pull in an image.  It's such a simple component it's probably not worth doing as a component.  The implementation is unlikely to change and even if it does it's a fairly simple search-and-replace.  I could also inline an SVG instead of using a file I suppose.

```
{% load static %}

<img id="{{ id }}" class="info-icon ml-4 filter dark:invert" src="{% static 'images/info.svg' %}">
```

**Reply by wrabit** [reply]
I wouldn't worry about any of it. Unless for some reason you needed to render this 100+ times I might consider benchmarking it and optimising for performance.
